client/common/BUILD.bazel

```diff
@@ -0,0 +1,36 @@
+load("//client/shared/dev:defaults.bzl", "ts_config", "ts_project")
+
+ts_config(
+    name = "tsconfig",
+    src = "tsconfig.json",
+)
+
+ts_project(
+    name = "common",
+    srcs = glob(
+        [
+            "*.ts",
+            "src/**/*",
```

Comment:
I think we should only accept things within `src/` and it should only be `*.ts` files. Anything that is not .ts probably belongs in the `data` and anything outside `src` is normally config that shouldn't be in the main project.

Reply:
Oh, but you're just copying the tsconfig? 🤔

Reply:
Yeah I'm copying the tsconfig.

Reply:
I'm not sure if we'll want to do that though, or if we'll even want the local `tsconfig.json` or I think we can just use the root one and nothing else.

Does the `outDir` cause any issues? I don't think we want that.

Reply:
I'm not sure if we want the local tsconfig in each directory, things like the `sourceRoot` and `baseUrl` do odd things that idk if we want (or at least don't need) in bazel.

I'd vote to do as little as possible and just have one `ts_config` in the root (using the tsconfig.bazel.json that I created for that purpose) and every `ts_project` only uses that one config. The `ts_project` macro can for now just hardcode that config and all the attributes that must align with it.